src/main/java/core/basesyntax/Calculate.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Calculate {
+    private static final String SUPPLY = "supply,";
+    private static final String TO_BUY = "buy,";
+    private static final String RESULT = "result,";
+
+    public String calculateBuySupplyData(String data) {
+        int toBuy = 0;
+        int toSupply = 0;
+        for (int i = 0; i < data.split(" ").length; i++) {
+            if (data.split(" ")[i].split(",")[0].equals("buy")) {
+                toBuy += Integer.parseInt(data.split(" ")[i].split(",")[1]);
+            } else {
+                toSupply += Integer.parseInt(data.split(" ")[i].split(",")[1]);
+            }
```

Comment:
- you should use constants instead of String literals
- define a separate variables  - separate variable for split array, separate variables for indexed elements